### PR TITLE
LPS-166651 Subtask Complete and Update status of case results in subtask

### PR DESCRIPTION
**Pull request by rafaelfs98**

### CODE REVIEW

**Comment by rafaelfs98** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/TestrayRouter.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ import Suite from './pages/Project/Suites/Suite';
 import SuiteForm from './pages/Project/Suites/SuiteForm';
 import SuiteOutlet from './pages/Project/Suites/SuiteOutlet';
 import Testflow from './pages/Testflow';
-import Subtasks from './pages/Testflow/Subtask';
+import Subtasks from './pages/Testflow/Subtask/Subtask';
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/TestrayRouter.tsx:64]: Rename `Subtask.tsx` to `index.tsx`

```suggestion
import Subtasks from './pages/Testflow/Subtask';
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	const {subtaskId} = useParams();
+	const {testrayTask} = useOutletContext<OutletContext>();
+
+	const {data: testraySubtask, mutate: mutateSubtask} = useFetch<
+		TestraySubTask
+	>(testraySubtaskImpl.getResource(subtaskId as string), (response) =>
+		testraySubtaskImpl.transformData(response)
+	);
+
+	const {data: testrayCaseResultData, mutate: mutateCaseResult} = useFetch<
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:57]:
```suggestion
	const {data: testraySubTaskCaseResultData, mutate: mutateCaseResult} = useFetch<
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	>(
+		`${testraySubtaskCaseResultImpl.resource}&filter=${searchUtil.eq(
+			'subtaskId',
+			subtaskId as string
+		)}&pageSize=100`,
+		(response) =>
+			testraySubtaskCaseResultImpl.transformDataFromList(response)
+	);
+
+	const testrayCaseResults = testrayCaseResultData?.items || [];
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:68]:
```suggestion
	const testraySubTaskCaseResults = testraySubTaskCaseResultData?.items || [];
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/types.ts [hunk 4]
```diff
@@ -202,13 +202,29 @@ export type TestrayRun = {
...
 	name: string;
+	r_taskToSubtasks_c_task: TestrayTask;
+	r_userToSubtasks_user: UserAccount;
 	score: number;
+	statusUpdateDate: string;
+	task: TestrayTask;
+	user: UserAccount;
+};
+
+export type TestraySubTaskCasesResult = {
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/types.ts:221]:
```suggestion
export type TestraySubTaskCaseResult = {
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+					title: `${testraySubtask?.name}`,
+				},
+			]);
+		});
+	}, [
+		setHeading,
+		testraySubtask?.name,
+		subtaskId,
+		testrayTask.id,
+		testrayTask?.name,
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:89]:
```suggestion
		testrayTask,
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+				{
+					category: i18n.translate('subtask'),
+					title: `${testraySubtask?.name}`,
+				},
+			]);
+		});
+	}, [
+		setHeading,
+		testraySubtask?.name,
+		subtaskId,
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:87]:
```suggestion
		testraySubtask
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+	const testrayCaseResults = testrayCaseResultData?.items || [];
+
+	useEffect(() => {
+		setTimeout(() => {
+			setHeading([
+				{
+					category: i18n.translate('task'),
+					path: `/testflow/${testrayTask.id}`,
+					title: `${testrayTask?.name}`,
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:76]:
```suggestion
					title: testrayTask.name,
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+				},
+				{
+					category: i18n.translate('subtask'),
+					title: `${testraySubtask?.name}`,
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:80]:
```suggestion
					title: testraySubtask.name,
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+	if (!testraySubtask) {
+		return <Loading />;
+	}
+
+	return (
+		<>
+			<SubtaskHeaderActions
+				caseResult={testrayCaseResults.map((caseResult) =>
+					Number(caseResult.caseResult?.id)
+				)}
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/Subtask.tsx:101]:
```suggestion
				caseResultIds={testrayCaseResults.map((caseResult) =>
					Number(caseResult.caseResult?.id)
				)}
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/SubtaskCompleteModal.tsx [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+import {withVisibleContent} from '../../../hoc/withVisibleContent';
+import {FormModalOptions} from '../../../hooks/useFormModal';
+import i18n from '../../../i18n';
+import yupSchema, {yupResolver} from '../../../schema/yup';
+import {TEST_STATUS} from '../../../util/constants';
+
+type SubtaskForm = {
+	dueStatus: string;
+	issue: string;
+};
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/SubtaskCompleteModal.tsx:29]: Use Type from Yup

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Pull request closed without merging**